### PR TITLE
PANGOLIN-3680: data table settings menu

### PR DESCRIPTION
**Pull request by kterry1**
#### Summary

The DataTableManager's table settings has been replaced with the new [DropdownMenu](http://localhost:9002/?path=/story/components-dropdowns-dropdownmenu--dropdownmenu-list-menu-content) component and the tooltip as been added.

### CODE REVIEW

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.tsx [hunk 1]
```diff
@@ -121,12 +122,10 @@ export type TDataTableSettingsProps = {
   managerTheme?: 'light' | 'dark';
 };
 
-/* The horizontal constraint is set on this container instead of the SelectInput
-because the input is always empty, and therefore doesn't take any space by itself
-but we want to keep enough space for the placeholder to be readable */
-const SelectContainer = styled.div`
-  min-width: ${designTokens.constraint4};
-`;
```
Comment (kterry1) [inline packages/components/data-table-manager/src/data-table-settings/data-table-settings.tsx:129]: No longer needed as we've removed the input option.

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.tsx [hunk 2]
```diff
@@ -223,23 +214,34 @@ const DataTableSettings = (props: TDataTableSettingsProps) => {
...
-              options={dropdownOptions}
-              iconLeft={<TableIcon color="neutral60" />}
-            />
-          </SelectContainer>
+            <Tooltip placement="left" title="Table settings">
+              <DropdownMenu
+                triggerElement={
+                  <IconButton icon={<ColumnsIcon />} label="list" />
+                }
+                menuHorizontalConstraint={4}
```
Comment (kterry1) [inline packages/components/data-table-manager/src/data-table-settings/data-table-settings.tsx:228]: `menuHorizontalContraint` with value of `3` would wrap the text in the dropdown so I upped it to `4` as that was the closest I could get to the design and ensuring no wrapping occurred.

Comment (ddouglasz) [reply packages/components/data-table-manager/src/data-table-settings/data-table-settings.tsx:228]: I think we need to tag @FilPob here to confirm to approve of this decision.

**Review by ddouglasz — COMMENTED** [review]
Awesome work 💯
Please remember to add a changeset 🙏🏾.

File: .changeset/slimy-kangaroos-hammer.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-uikit/data-table-manager': minor
+---
+
+feat(data-table-settings): added new dropDownMenu to dataTableManager
```
Comment (ddouglasz) [inline .changeset/slimy-kangaroos-hammer.md:5]:
```suggestion
replace the dataTableManager SelectInput with the new dropDownMenu component
```

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.tsx (hunk 2 above)
Comment (CarlosCortizasCT) [reply packages/components/data-table-manager/src/data-table-settings/data-table-settings.tsx]: I think that's a good option, but bear in mind this adjustment works for english translation but might not work for other languages.
For instance, if we use portuguese, the text will be wrapped anyway.

**Comment by FilPob** [comment]
@kterry1 Could you please try to set the `menuHorizontalConstraint` to `auto` ?  By that the menu should wrap around the content regardless of it's length

**Comment by CarlosCortizasCT** [comment]
I think we should not merge this PR until [this issue](https://github.com/commercetools/ui-kit/issues/2786) is fixed.
If we do, we will be releasing a change which will contain a know bug and will downgrade our user experience.

**Comment by kterry1** [comment]
> @kterry1 Could you please try to set the `menuHorizontalConstraint` to `auto` ? By that the menu should wrap around the content regardless of it's length

@FilPob done!

**Comment by kterry1** [comment]
> I think we should not merge this PR until [this issue](https://github.com/commercetools/ui-kit/issues/2786) is fixed. If we do, we will be releasing a change which will contain a know bug and will downgrade our user experience.

@CarlosCortizasCT Is Shield planning on working on this, or is this something Pangolin Team needs to pull in?

**Comment by CarlosCortizasCT** [comment]
> > I think we should not merge this PR until [this issue](https://github.com/commercetools/ui-kit/issues/2786) is fixed. If we do, we will be releasing a change which will contain a know bug and will downgrade our user experience.
> 
> @CarlosCortizasCT Is Shield planning on working on this, or is this something Pangolin Team needs to pull in?

Hi @kterry1 

We will work on it and we will try to do it this week ([reference](https://commercetools.slack.com/archives/C06T7EH7R97/p1713799830853469?thread_ts=1713796485.710169&cid=C06T7EH7R97)).

**Comment by CarlosCortizasCT** [comment]
@kterry1 We've just merged [a fix](https://github.com/commercetools/ui-kit/pull/2804) for the issues I commented on Monday so it's good to move forward with this one.

**Review by FilPob — APPROVED** [review]

**Review by CarlosCortizasCT — APPROVED** [review]
Looks good to me 👍 

Thanks Kevin!

**Review by chloe0592 — APPROVED** [review]

**Pull request merged**